app/controllers/application_controller.rb

```diff
@@ -9,4 +11,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  # Class accessor for an instance of RedirectMap
+  def self.redirect_map
+    @redirect_map ||= RedirectMap.instance
+  end
+
+  # Instance accessor for the redirect map. This is used in the :before_action
+  # hook.
+  def redirect_map
+    self.class.redirect_map
+  end
```

Comment:
If nothing beyond the hook needs this, seems like just calling the class method directly below would be fine.